src/Components/ADT3DSceneBuilder/Internal/VisualRuleForm/Internal/ConditionsList.tsx

```diff
@@ -41,63 +65,109 @@ const ConditionsList: React.FC<IConditionsListProps> = (props) => {
     const classNames = getClassNames(styles, {
         theme: useTheme()
     });
-    const getOverflowMenuItems = (
-        conditionId: string
-    ): IContextualMenuItem[] => [
-        {
-            key: `${conditionId}-edit-menu-item`,
-            text: t('3dSceneBuilder.visualRuleForm.editCondition'),
-            iconProps: {
-                iconName: 'Edit'
-            },
-            onClick: () => {
-                alert('Edit clicked');
+
+    const [calloutInfo, setCalloutInfo] = useState<CalloutInfo>({
+        calloutType: CalloutInfoType.inactive,
+        isOpen: false,
+        selectedCondition: null,
+        selectedTarget: null
+    });
+
+    // Callbacks
+    const handleOpenNewConditionFlyout = useCallback(() => {
+        setCalloutInfo({
+            calloutType: CalloutInfoType.create,
+            isOpen: true,
+            selectedCondition: getDefaultCondition(valueRangeType),
+            selectedTarget: `#${LIST_KEY}`
+        });
+    }, [valueRangeType]);
+
+    const handleDismissFlyout = useCallback(() => {
+        setCalloutInfo({
+            calloutType: CalloutInfoType.inactive,
```

Comment:
can we create a "defaultInfo" object? It seems like that could be used in both the initial setState as well as the dismiss function. I'm guessing we want those to be the same values so we want dismiss to go back to the same as the initial state.